lib/fog/core/cache.rb

```diff
@@ -109,34 +109,81 @@ def self.load(model_klass, service)
 
       raise CacheNotFound if cache_files.empty?
 
-      attributes = cache_files.map do |path|
-        load_cache(path)[:attrs]
+      loaded = cache_files.map do |path|
+        if valid_for_load?(path)
+          model_klass = const_get_compat(load_cache(path)[:model_klass])
```

Comment:
Does this need to be repeated? Shouldn't it be the same as the passed-in model_klass value that is already set in the method scope?